### PR TITLE
Fix string face

### PR DESCRIPTION
**Pull request by yuutayamada**
This PR is improved #44 with some tests.
(If test is not enough, I can wait)

### CODE REVIEW

**Comment by pigmej** [comment]
Awesome :)

Does everything work as expected ? No more unknown / wrong behaviours ?

**Comment by yuutayamada** [comment]
> Does everything work as expected ? No more unknown / wrong behaviours ?

The character delimiter works almost I expected.
But when I input '^@'(control character) as character, I got flycheck's error
"invalid character constant".

The control characters' highlight are working, but should I omit
those highlight? I implemented it because according to
http://nim-lang.org/docs/manual.html#lexical-analysis-character-literals
the character type is single byte...

Also I have little bit concern about typing single quote because
I changed syntax-table of single quote and I seem this affected
electric-pair mode. (should nim-mode provide similar function
for single quote?)

**Comment by reactormonk** [comment]
Does anyone use electric-pair-mode that could comment on that part? I don't personally use it (so far).

**Comment by yuutayamada** [comment]
Sorry, I didn't notice that Nim can use following form.

```nim
echo """""foo"""""""""""
```

Above commits apply this string delimiter's form.

**Comment by yuutayamada** [comment]
> Does anyone use electric-pair-mode that could comment on that part? I don't personally use it (so far).

electric-pair-mode is kind of smart self-insert-command. For example, insert a single quote when
after number for number type specification, otherwise insert two single quotes for character delimiter.
The number of quote varies depends on context.

File: tests/test-syntax.el [hunk 1]
```diff
@@ -54,12 +119,32 @@
...
+ (test-characters
+  "should highlight characters correctly"
+  (test-concat-dir "tests/syntax/char.nim"))
+
+ ;; Number
  (test-faces-by-range
-  "should highlight comment correctly after char of double quote"
-  (test-concat-dir "tests/syntax/string.nim")
-  '(((119 . 165) . font-lock-comment-face))))
+  "should not highlight numbers by string-face"
```
Comment (reactormonk) [inline tests/test-syntax.el]: I don't like the idea of `not` tests. Could you just expand it to a few tests for `number-face`?

**Comment by reactormonk** [comment]
I don't think it's necessary to correctly handle control characters in the sources. If you use these instead of `\n` or `\x23`, your code is bad and you should feel bad.

File: tests/test-syntax.el (hunk 1 above)
Comment (yuutayamada) [inline tests/test-syntax.el]: Sure. I changed the test.

**Comment by yuutayamada** [comment]
> I don't think it's necessary to correctly handle control characters in the sources. If you use these instead of \n or \x23, your code is bad and you should feel bad.

I agree with you. So should I omit control characters highlight?

**Comment by reactormonk** [comment]
You can keep it in if it doesn't make the code more complicated. But if it conflicts with anything (electric apparently), toss it.

**Comment by yuutayamada** [comment]
I updated the character delimiter's regex. The test is still working and I seem 
highlighting is working well. (I just checked syntax highlight on test file.)

**Pull request merged**